mimic/rest/identity_api.py

```diff
@@ -669,6 +688,521 @@ def list_tenants(self, request):
+        any endpoint templates assigned to it for success.
+
+        Note: Part of the OS-KSADM extension.
+        """
+        x_auth_token = request.getHeader(b"x-auth-token")
+        if x_auth_token is None:
+            return json.dumps(unauthorized("Authentication required", request))
+
+        try:
+            self.core.remove_external_api(
+                service_id
+            )
+        except ServiceDoesNotExist:
+            return json.dumps(
+                not_found(
+                    "Service not found. Unable to remove.",
+                    request))
+        except ServiceHasTemplates:
+            return json.dumps(
+                conflict(
+                    "Service still has endpoint templates.",
+                    request))
+        else:
+            request.setResponseCode(204)
+            return b''
+
+    @app.route('/v2.0/OS-KSCATALOG/endpointTemplates', methods=['GET'])
+    def list_endpoint_templates(self, request):
+        """
+        List the available endpoint templates.
+
+        Reference: http://developer.openstack.org/api-ref-identity-v2-ext.html
+
+        Note: Marker/Limit capability not implemented here.
+        """
+        x_auth_token = request.getHeader(b"x-auth-token")
+        if x_auth_token is None:
+            return json.dumps(unauthorized("Authentication required", request))
+
+        # caller may provide a specific API to list by setting the
+        # serviceid header
+        external_apis_to_list = []
+        service_id = request.getHeader(b'serviceid')
+        if service_id is not None:
+            external_apis_to_list = [service_id.decode('utf-8')]
+        else:
+            external_apis_to_list = [
+                api_id
+                for api_id in self.core.get_external_apis()
+            ]
+
+        try:
+            data = []
+            request.setResponseCode(200)
+            for api_id in external_apis_to_list:
+                api = self.core.get_external_api(api_id)
+                for endpoint_template in api.list_templates():
+                    data.append(endpoint_template.serialize())
+
+            return json.dumps(
+                {
+                    "OS-KSCATALOG": data,
+                    "OS-KSCATALOG:endpointsTemplates_links": []
+                }
+            )
+        except ServiceDoesNotExist:
+            request.setResponseCode(404)
+            return json.dumps(not_found(
+                "Unable to find the requested API",
+                request))
+
+    @app.route('/v2.0/OS-KSCATALOG/endpointTemplates', methods=['POST'])
+    def add_endpoint_templates(self, request):
+        """
+        Add an API endpoint template to the system. By default the API
+        described by the template will disabled for all users.
+
+        Reference: http://developer.openstack.org/api-ref-identity-v2-ext.html
+
+        Note: Either the service-id must be specified in the header or
```

Comment:
Not worth changing it in this branch, but the correct way to format this with a ReST docstring is `.. note::`, as documented [here](http://sphinx.readthedocs.io/en/stable/markup/para.html?highlight=note#directive-note).